src/zerocoin.cpp

```diff
@@ -564,9 +564,9 @@ bool CheckZerocoinTransaction(const CTransaction &tx,
                               CZerocoinTxInfo *zerocoinTxInfo)
 {
     // nHeight have special mode which value is INT_MAX so we need this.
-    int realHeight;
+    int realHeight = params.nSigmaStartBlock;
 
-    {
+    if(!(isVerifyDB && nHeight == INT_MAX)) {
```

Comment:
I think this seem like a hack. After investigate I found the expression `isVerifyDB && nHeight == INT_MAX` will be `true` only when loading wallet. Can we do something like this instead?

```cpp
bool isWalletCheck = (isVerifyDB && nHeight == INT_MAX);
```

and then change bottom logics to something like this without initialize `realHeight` with `params.nSigmaStartBlock`:

```cpp
if (!isWalletCheck && realHeight > params.nSigmaStartBlock + params.nZerocoinV2MintGracefulPeriod) {
```

```cpp
if (!isWalletCheck && realHeight > params.nSigmaStartBlock + params.nZerocoinV2SpendGracefulPeriod) {
```

Reply:
The way it is implemented now saves re-entering the cs_main. I am keeping this property.